pkg/http/handler/timeout_test.go

```diff
@@ -24,15 +25,74 @@ import (
 	"time"
 )
 
+func TestTimeoutWriterAllowsForAdditionalWrites(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.WriteHeader(http.StatusOK)
+	handler.TimeoutAndWriteError("error")
+	if _, err := io.WriteString(handler, "test"); err != nil {
+		t.Fatalf("handler.Write() = %v, want no error", err)
+	}
+
+	if got, want := recorder.Code, http.StatusOK; got != want {
+		t.Errorf("recorder.Status = %d, want %d", got, want)
+	}
+	if got, want := recorder.Body.String(), "test"; got != want {
+		t.Errorf("recorder.Body = %s, want %s", got, want)
+	}
+}
+
+func TestTimeoutWriterDoesntFlushAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.TimeoutAndWriteError("error")
+	handler.Flush()
+
+	if got, want := recorder.Flushed, false; got != want {
+		t.Errorf("recorder.Flushed = %t, want %t", got, want)
+	}
+}
+
+func TestTimeoutWriterFlushesAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.Flush()
+
+	if got, want := recorder.Flushed, true; got != want {
+		t.Errorf("recorder.Flushed = %t, want %t", got, want)
+	}
+}
+
+func TestTimeoutWriterErrorsWriteAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.TimeoutAndWriteError("error")
+	if _, err := handler.Write([]byte("hello")); err != http.ErrHandlerTimeout {
+		t.Errorf("ErrHandlerTimeout got %v, want: %s", err, http.ErrHandlerTimeout)
+	}
+
+	handler.WriteHeader(http.StatusOK)
+	if got := recorder.Code; got == http.StatusOK {
+		t.Errorf("WriteHeader got %d want %d", got, 0)
```

Comment:
```suggestion
		t.Errorf("WriteHeader got %d want %d", got, http.StatusOK)
```

Reply:
If we got `http.StatusOK` then it's an error. The test is to make sure that after timeout, attempting to write a different header will not override the timeout.

Reply:
As the test was just from a file move, and we didn't modify it. We didn't go through it to make sure it was reporting the right thing or not. 

Removing that part of the test as `ResponseRecorder` doesn't allow writing header twice. It's not actually testing anything.